src/Akeneo/Connectivity/Connection/back/Infrastructure/EventSubscriber/ReadProductsEventSubscriber.php

```diff
@@ -47,24 +49,65 @@ public static function getSubscribedEvents(): array
      */
     public function saveReadProducts(ReadProductsEvent $event): void
     {
-        if (!$this->connectionContext->isCollectable()) {
-            return;
-        }
-        if (0 === count($event->productIds())) {
+        if (0 === $event->getCount()) {
             return;
         }
+        $connectionCode = $this->getConnectionCodeByReadProductsEvent($event);
 
-        $connection = $this->connectionContext->getConnection();
-        if (FlowType::DATA_DESTINATION !== (string) $connection->flowType()) {
+        if (!$this->connectionIsAuditable($connectionCode, $event->isEventsApi())) {
             return;
         }
 
         $this->updateDataDestinationProductEventCountHandler->handle(
             new UpdateDataDestinationProductEventCountCommand(
-                (string) $connection->code(),
+                $connectionCode,
                 HourlyInterval::createFromDateTime(new \DateTimeImmutable('now', new \DateTimeZone('UTC'))),
-                count($event->productIds())
+                $event->getCount()
             )
         );
     }
+
+    /**
+     * @throws \LogicException
+     */
+    private function getConnectionCodeByReadProductsEvent(ReadProductsEvent $event): string
+    {
+        if ($event->isEventsApi()) {
+            $connectionCode = $event->getConnectionCode();
+            if ($connectionCode === null) {
+                throw new \LogicException('This connection code is empty on a event api.');
```

Comment:
```suggestion
                throw new \LogicException('You must provide a connection code through the event when using the events API.');
```
https://www.notion.so/akeneo/Fixing-Error-Messages-437e5a74aa17409db88c8bfcafbd4718